ui/pages/routes/routes.component.js

```diff
@@ -650,7 +672,9 @@ export default class Routes extends Component {
           ///: END:ONLY_INCLUDE_IN
         }
         <Box className="main-container-wrapper">
-          {isLoading ? <Loading loadingMessage={loadMessage} /> : null}
+          {shouldShowLoadingPage ? (
+            <Loading loadingMessage={loadMessage} />
+          ) : null}
```

Comment:
nit: 
```suggestion
          {isLoadingShown ?? (<Loading loadingMessage={loadMessage} />)}
```

Reply:
My understanding of the nullish operator is that is will return the left hand side of the operation if it evaluates to false. Since this is a boolean I think this would result in false. The ternary returns null which means nothing will be rendered.

<img width="228" alt="Screenshot 2023-11-01 at 3 11 34 PM" src="https://github.com/MetaMask/metamask-extension/assets/22918444/29f5b2f1-c5da-44db-af75-1ece9bb6a33d">

Reply:
for the suggestion, it is a JSX specific notation https://react.dev/learn/conditional-rendering#logical-and-operator-. I think it should work